From 117b31c362f95e07c95fb1bf4a2e4fbc01471267 Mon Sep 17 00:00:00 2001
From: Nickey Yang <nickey.yang@rock-chips.com>
Date: Mon, 27 Aug 2018 15:56:23 -0400
Subject: [PATCH] BACKPORT: media: cec: move compat_ioctl handling to cec-api.c

All the CEC ioctls are compatible, and they are only implemented
in one driver, so we can simply let this driver handle them
natively.

Change-Id: I04136c171777d5ab23760e20162f0a6197090ca8
Signed-off-by: Arnd Bergmann <arnd@arndb.de>
Reviewed-by: Hans Verkuil <hans.verkuil@cisco.com>
Signed-off-by: Mauro Carvalho Chehab <mchehab+samsung@kernel.org>
Signed-off-by: Nickey Yang <nickey.yang@rock-chips.com>
(cherry picked from commit e6c8320648a27d2f43bf56c6215fa48d85bf47c6)
---
 drivers/media/cec/cec-api.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/media/cec/cec-api.c b/drivers/media/cec/cec-api.c
index 3dba3aa34a43..7e87c43b1bd9 100644
--- a/drivers/media/cec/cec-api.c
+++ b/drivers/media/cec/cec-api.c
@@ -661,6 +661,7 @@ const struct file_operations cec_devnode_fops = {
 	.owner = THIS_MODULE,
 	.open = cec_open,
 	.unlocked_ioctl = cec_ioctl,
+	.compat_ioctl = cec_ioctl,
 	.release = cec_release,
 	.poll = cec_poll,
 	.llseek = no_llseek,
-- 
2.35.3

